### DISCUSSION TITLE
Enhancement Request: Use PulseAudio

### DISCUSSION THREAD

**Discussion by sromo006** [vicwomg/pikaraoke#797] [General]
Instead of using alsa for the audio subsystem, as I see the note about latency and audio quality, upgrade to use PulseAudio for more options around loopback configurations and audio settings when using a hat such as hifiberry. I think this would be a great option for USB mics or mixed mics where the desire is for all of the output to come out of the same HDMI output.  With the an audio card in the Pi you wont have the quality issues but just need a better way to control latency and loopback.

**Comment by alcalda** [comment]
I would rather target for Pipewire as Soundserver. It unifies Alsa, Pulseaudio and JACK API and has less latency than Pulseaudio.
As Raspbian Buster still sticks with the 0.2.5 version, it would also require either Bullseye or to use testing Repositories.

**Comment by vaccinemedia** [comment]
I have a PiSound interface and having a look at using Patchbox OS and MODEP for adding effects to a mic input with zero latency. Patchbox and MODEP uses Jack Audio. So far I have MODEP running on startup with some reverb being applied to the mic input which works great. My only issue I have now is that Pikaraoke does not use Jack Audio so I don't know how to get both playing at the same time. Patchbox is set up in a way that allows multiple apps to share Jack Audio at the same time so it would be good to have a solution which would allow for Karaoke plus mic inputs with effects to be mixed and played together. From experience, HDMI devices have their own latency issues so just the default output would be great if possible.
